src/core/common/permissions/validateRoleChange.ts

```diff
@@ -0,0 +1,73 @@
+import { isSiteModerationScoped } from "./isSiteModerationScoped";
+import { User, UserRole } from "./types";
+
+type Moderator = User & {
+  role: "MODERATOR";
+};
+
+type LTEModerator = User & {
+  role: "MODERATOR" | "STAFF" | "MEMBER" | "COMMENTER";
+};
+
+const isModerator = (user: User): user is Moderator => {
+  return user.role === "MODERATOR";
+};
+
+const isLTEModerator = (user: User): user is LTEModerator => {
+  return user.role !== "ADMIN";
+};
+
+const validateModeratorRoleChange = (
+  viewer: Readonly<Moderator>,
+  user: Readonly<LTEModerator>,
+  role: UserRole,
+  scoped: boolean
+) => {
+  // Org mods may promote users to site moderators within their scope,
+  // but only if the user < org mod
+  const viewerIsOrgMod = !isSiteModerationScoped(viewer.moderationScopes);
+  const roleIsSiteMod = role === "MODERATOR" && scoped;
+  const userLTOrgMod =
+    isSiteModerationScoped(user.moderationScopes) || user.role !== "MODERATOR";
+
+  if (viewerIsOrgMod && userLTOrgMod && roleIsSiteMod) {
+    return true;
+  }
+
+  return false;
+};
+
+export const validateRoleChange = (
+  viewer: Readonly<User>,
+  user: Readonly<User>,
+  role: UserRole,
+  /**
+   * TODO: this optional argument that is only relelvant
```

Comment:
Can you either remove this todo comment or update to remove typos for future reference on what we should do to change this?